src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const char of actions) {
+    if (char.type === 'addProperties') {
+      for (const exData in char.extraData) {
+        state[exData] = char.extraData[exData];
+      }
+    }
+
+    if (char.type === 'removeProperties') {
+      for (const keyRemove of char.keysToRemove) {
+        if (state.hasOwnProperty(keyRemove)) {
```

Comment:
if a given `keyToRemove` doesn't exist on `state`, we won't delete anything, thus there is no need to check if one contains it